src/network/networkd-routing-policy-rule.h

```diff
@@ -0,0 +1,68 @@
+#pragma once
+
+/***
+  This file is part of systemd.
+
+  Copyright 2017 Susant Sahani
+
+  systemd is free software; you can redistribute it and/or modify it
+  under the terms of the GNU Lesser General Public License as published by
+  the Free Software Foundation; either version 2.1 of the License, or
+  (at your option) any later version.
+
+  systemd is distributed in the hope that it will be useful, but
+  WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+  Lesser General Public License for more details.
+
+  You should have received a copy of the GNU Lesser General Public License
+  along with systemd; If not, see <http://www.gnu.org/licenses/>.
+***/
+
+#include <inttypes.h>
+#include <stdbool.h>
+
+#include "in-addr-util.h"
+
+typedef struct RoutingPolicyRule RoutingPolicyRule;
+
+#include "networkd-link.h"
+#include "networkd-network.h"
+
+typedef struct Network Network;
+typedef struct Link Link;
+typedef struct NetworkConfigSection NetworkConfigSection;
+
+struct RoutingPolicyRule {
+        Network *network;
+        Link *link;
+        NetworkConfigSection *section;
+
+        uint8_t tos;
+
+        uint32_t table;
+        uint32_t fwmark;
+        uint32_t fwmask;
+        uint32_t priority;
+
+        int family;
+        unsigned char prefixlen;
```

Comment:
@ssahani well, but @keszybz point is really: most objects networkd manages are in one way or another associated with an interface, and we know for each interface if we consider ourselves as "owner" of it, or if we let other subsystems manage it. The IP rules logic doesn't really fit in this scheme: you could run NM and networkd side-by-side, and they would necessarily interfere with each others IP rules settings, because the rules have no explicit ownership and you cannot really construct ownership for them either in the kernel afaics...

The clean way out to me appears to be to always write to /run the rules we installed in the kernel, and then consider those owned by networkd, and leave all others unaffected.